### PR TITLE
Add Linux Support (Optional PR).

### PR DESCRIPTION
**Pull request by Weather-OS**
Merging this PR is purely optional as 99.9999% of BetterRTX users don't tend to run GNU/Linux.

Everything is tested and the script works perfectly fine.

### CODE REVIEW

**Comment by jasonjgardner** [comment]
AFAIK, ray tracing is not supported on Linux or in the Education Edition: https://help.minecraft.net/hc/en-us/articles/4408865164173-Minecraft-with-Ray-Tracing-and-Advanced-Graphics-FAQ

I have not tested this personally though. Does it work? @Weather-OS

**Comment by Weather-OS** [comment]
> AFAIK, ray tracing is not supported on Linux or in the Education Edition: https://help.minecraft.net/hc/en-us/articles/4408865164173-Minecraft-with-Ray-Tracing-and-Advanced-Graphics-FAQ
> 
> I have not tested this personally though. Does it work? @Weather-OS

Yes, Indeed it does. Although with certain limitations.
1) You can only run Minecraft RTX within Minecraft Education, since the Windows 10 edition is a UWP application and sophisticated UWP applications currently don't run under Wine or Linux. Minecraft Education though, is based on Win32 and it works perfectly fine as a Minecraft Bedrock Edition as long as you don't mind not having access to the market place or online features.

2) You can only sign into Minecraft Education prior to version 1.19.x. This is fine though as 1.18.x introduced RenderDragon which itself introduced Minecraft RTX to Minecraft Education. While it is possible to bypass the login screen on versions beyond 1.19.x, it's taboo and I don't really recommend it.

3) Hardware accelerated ray tracing is indeed supported on Linux on all Vendors that provide such support (Intel, NVIDIA and AMD to name a few). Though this is only exposed through Vulkan, it is possible to run DXR 1.1 Applications through a [translation layer](https://github.com/HansKristian-Work/vkd3d-proton)

If you still have your doubts, I've made a full video covering how to run Minecraft RTX on Linux and I've made a [showcase video](https://www.youtube.com/watch?v=dIcGpHjiUAU) as well.

**Comment by jasonjgardner** [comment]
That's impressive. Can you explain how and why the RTXStub file is patched? For Vulkan compatibility I assume? Each preset will have a different `RTXStub.material.bin` file, so replacing it could interfere with presets.

**Comment by Weather-OS** [comment]
> That's impressive. Can you explain how and why the RTXStub file is patched? For Vulkan compatibility I assume? Each preset will have a different `RTXStub.material.bin` file, so replacing it could interfere with presets.

The patched RTXStub is a single stub that’s based on BetterRTX Experimental version 1.1.
It’s there for people who use an Intel Arc graphics card or don’t want to use AMD’s Proprietary drivers on Linux.

File: v1/BetterRTX_Installer.sh [hunk 1]
```diff
@@ -0,0 +1,433 @@
...
+
+function _checkForLolcat {
+    if [ -f "/bin/lolcat" ]; then
+        return
+    else
+        false
+    fi
+}
+
+function _checkForBinary {
```
Comment (joltzbronzer) [inline v1/BetterRTX_Installer.sh:268]: Maybe use `pkg-config` to check for binaries? Some distributions are immutable and have multiple end points for their binaries.

**Comment by ABUCKY0** [comment]
I'm going to close this PR since we cannot commit to keeping this updated and working with Linux.

**Pull request closed without merging**